lua/neogit/popups/init.lua

```diff
@@ -0,0 +1,109 @@
+local M = {}
+
+---@param name string
+---@param f nil|fun(create: fun(...)): any
+--- Creates a curried function which will open the popup with the given name when called
+function M.open(name, f)
+  f = f or function(c)
+    c()
+  end
+
+  return function()
+    local ok, value = pcall(require, "neogit.popups." .. name)
+    if ok then
+      assert(value, "popup is not nil")
+      assert(value.create, "popup has create function")
+
+      f(value.create)
+    else
+      local notification = require("neogit.lib.notification")
+      notification.create(string.format("No such popup: %q", name), vim.log.levels.ERROR)
+    end
+  end
+end
+
+--- Returns an array useful for creating mappings for the available popups
+---@return table<string, Mapping>
+function M.mappings_table()
+  local config = require("neogit.config")
+  local async = require("plenary.async")
+  return {
+    {
+      "HelpPopup",
+      "Help",
+      M.open("help", function(f)
+        f {
+          use_magit_keybindings = config.values.use_magit_keybindings,
+        }
+      end),
+    },
+    { "DiffPopup", "Diff", M.open("diff") },
+    { "PullPopup", "Pull", M.open("pull") },
+    {
+      "RebasePopup",
+      "Rebase",
+      M.open("rebase", function(f)
+        local status = require("neogit.status")
+        local line = status.status_buffer:get_current_line()
+        f { line[1]:match("^(%x%x%x%x%x%x%x+)") }
+      end),
+    },
+    { "MergePopup", "Merge", M.open("merge") },
+    { "PushPopup", "Push", M.open("push") },
+    { "CommitPopup", "Commit", M.open("commit") },
+    { "LogPopup", "Log", M.open("log") },
+    {
+      "CherryPickPopup",
+      "Cherry Pick",
+      {
+        "nv",
+        M.open(
+          "cherry_pick",
+          async.void(function(f)
```

Comment:
Can you verify that this needs to be `async.void`? I'm not sure that it's needed, especially considering the revert popup isn't, and... they're real similar.

Reply:
I don't think it needs to either, it was how it was at present.

The difference is that async.void may run on the next microtick, though not exactly sure. It should be immediate unless it needs that microframe delay for getting the visual selection. Will remove and tset

Reply:
Yea, pretty sure I added that because I had no idea how async stuff worked when I did.

Reply:
I tested it and it works without.

There is an error on master when doing `AA` on a visual selection. More testing (again) would catch it

```sh
Error executing vim.schedule lua callback: ...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:18: The coroutine failed with this message: ...rs/teiroberts/dev/nvim/neogit/lua/neogit/lib/git/cli.lua:749: invalid value (table) at index 8 in table for 'concat'
stack traceback:
	[C]: in function 'error'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:18: in function 'callback_or_next'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:45: in function <...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:44>

```